conjurapi/authn/auth_token_test.go

```diff
@@ -88,41 +85,6 @@ func TestTokenV4_Parse(t *testing.T) {
 		So(expired_token.ShouldRefresh(), ShouldEqual, true)
 	})
 
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-	}
-
-	httpClient := http.Client{Transport: tr}
-
-	authUrl := fmt.Sprintf("%s/authn/users/%s/authenticate",
-		os.Getenv("CONJUR_V4_APPLIANCE_URL"),
-		url.PathEscape(os.Getenv("CONJUR_V4_AUTHN_LOGIN")))
-
-	req, err := http.NewRequest("POST",
-		authUrl,
-		bytes.NewBuffer([]byte(os.Getenv("CONJUR_V4_AUTHN_API_KEY"))))
-
-	if err != nil {
-		log.Printf("Failed creating request to authenticate: %s\n", err)
-		log.Println("Cannot continue.")
-		return
-	}
-
-	req.Header.Set("Accept", "*/*")
-	res, err := httpClient.Do(req)
-	if err != nil {
-		log.Printf("Failed creating request to authenticate: %s\n", err)
-		log.Println("Cannot continue.")
-		return
-	}
-
-	if res.StatusCode != http.StatusOK {
-		log.Printf("Received %d response when authenticating.\n", res.StatusCode)
-		log.Println("Cannot continue.")
-	}
-
-	new_token_bytes, _ := ioutil.ReadAll(res.Body)
```

Comment:
why is this whole section removed? I'm not sure what is lost by mocking the new token, vs actually retrieving one via this sequence of http requests

Reply:
i thought of it like so:
+ an ad-hoc call to the api is overkill and forces this part of the code to need a v4 appliance to test
+ the token format is well defined so there's no loss in generality by using mocks
+ i think it's worth having these tests be well scoped. in this case to parsing a token and nothing more